test/system/src/test/java/io/pravega/test/system/ReadWriteAndScaleWithFailoverTest.java

```diff
@@ -0,0 +1,488 @@
+            readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder().startingTime(0).build(),
+                    Collections.singleton(STREAM_NAME));
+            log.info("Reader group name: {}, Reader group scope: {}, Online readers: {}",
+                    readerGroupManager.getReaderGroup(readerGroupName).getGroupName(), readerGroupManager
+                            .getReaderGroup(readerGroupName).getScope(), readerGroupManager
+                            .getReaderGroup(readerGroupName).getOnlineReaders());
+
+            //create readers
+            log.info("Creating {} readers", NUM_READERS);
+            List<EventStreamReader<Long>> readerList = new ArrayList<>(NUM_READERS);
+            log.info("Scope that is seen by readers {}", scope);
+            //start reading events
+            for (int i = 0; i < NUM_READERS; i++) {
+                log.info("Starting reader: {}, with id: {}", i, readerName + i);
+                final EventStreamReader<Long> reader = clientFactory.createReader(readerName + i,
+                        readerGroupName,
+                        new JavaSerializer<Long>(),
+                        ReaderConfig.builder().build());
+                readerList.add(reader);
+            }
+
+            // start writing asynchronously
+            List<CompletableFuture<Void>> writerFutureList = writerList.stream().map(writer ->
+                    startWriting(eventData, writer, stopWriteFlag, eventWriteCount)).collect(Collectors.toList());
+
+            //start reading asynchronously
+            List<CompletableFuture<Void>> readerFutureList = readerList.stream().map(reader ->
+                    startReading(eventsReadFromPravega, eventWriteCount, eventReadCount, stopReadFlag, reader))
+                    .collect(Collectors.toList());
+
+            //run the failover test before triggering scaling
+            performFailoverTest();
+
+            //increase the number of writers to trigger scale
+            List<EventStreamWriter<Long>> writerList1 = new ArrayList<>(ADD_NUM_WRITERS);
+            log.info("Writers writing in the scope {}", scope);
+            for (int i = 0; i < ADD_NUM_WRITERS; i++) {
+                log.info("Starting writer{}", i);
+                final EventStreamWriter<Long> writer1 = clientFactory.createEventWriter(STREAM_NAME,
+                        new JavaSerializer<Long>(),
+                        EventWriterConfig.builder().retryAttempts(10).build());
+                writerList1.add(writer1);
+            }
+
+            //start writing asynchronoulsy with newly created writers
+            List<CompletableFuture<Void>> writerFutureList1 = writerList1.stream().map(writer ->
+                    startWriting(eventData, writer, stopWriteFlag, eventWriteCount)).collect(Collectors.toList());
+
+
+            //bring the instances back to 3 before performing failover
+            controllerInstance.scaleService(3, true);
```

Comment:
This is not required, if we remove performFailoverTest() as per the above comment.